src/Tabs/Tabs.js

```diff
@@ -7,7 +8,7 @@ export class Tabs extends Component {
     super(props);
 
     this.state = {
-      id: 0
+      id: Number(props.selectedIndex) || 0
```

Comment:
you need to observe selectedIndex and update state if prop will be changed (later)